pkg/i2gw/providers/istio/converter.go

```diff
@@ -583,6 +583,10 @@ func (c *converter) convertVsHTTPRoutes(virtualService metav1.ObjectMeta, istioH
 		// set istio hostnames as is, without extra filters. If it's not a fqdn, it would be rejected by K8S API implementation
 		hostnames := make([]gatewayv1.Hostname, 0, len(allowedHostnames))
 		for _, host := range allowedHostnames {
+			// '*' is valid in istio, but not in HTTPRoute
+			if host == "*" {
```

Comment:
I left out validation to have less conversion logic and to defer this to the validation in the gw api implementation.

But having validation here is also quite fine. Could we make it a proper one then (from the gw api [HTTPRoute spec](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.HTTPRoute))
```
Valid values for Hostnames are determined by RFC 1123 definition of a hostname with 2 notable exceptions:

IPs are not allowed.
A hostname may be prefixed with a wildcard label (*.). The wildcard label must appear by itself as the first label.
```

1. to have a separate fun for the validation of the hostname
2. log if the hostname is discarded during the conversion